src/main/java/com/wuest/prefab/gui/controls/ExtendedButton.java

```diff
@@ -2,23 +2,85 @@
 
 import com.mojang.blaze3d.systems.RenderSystem;
 import com.wuest.prefab.Utils;
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
 import net.minecraft.client.MinecraftClient;
 import net.minecraft.client.gui.DrawableHelper;
+import net.minecraft.client.gui.screen.narration.NarrationMessageBuilder;
+import net.minecraft.client.gui.screen.narration.NarrationPart;
 import net.minecraft.client.gui.widget.ButtonWidget;
+import net.minecraft.client.gui.widget.PressableWidget;
 import net.minecraft.client.render.GameRenderer;
 import net.minecraft.client.util.math.MatrixStack;
 import net.minecraft.text.LiteralText;
 import net.minecraft.text.MutableText;
 import net.minecraft.text.Text;
+import net.minecraft.text.TranslatableText;
 import org.jetbrains.annotations.Nullable;
 
-public class ExtendedButton extends ButtonWidget {
+import java.util.ArrayList;
+
+public class ExtendedButton extends PressableWidget {
 	public float fontScale = 1;
 	private final String label;
+	private static final ButtonWidget.TooltipSupplier tooltipSupplier = (button, matrices, mouseX, mouseY) -> {};
+	private final PressAction pressAction;
+	private final boolean shouldRenderTooltip;
 
 	public ExtendedButton(int xPos, int yPos, int width, int height, Text displayString, PressAction handler, @Nullable String label) {
-		super(xPos, yPos, width, height, displayString, handler);
+		super(xPos, yPos, width, height, displayString);
+		pressAction = handler;
+		this.label = label;
+		this.shouldRenderTooltip = true;
+	}
+
+	public ExtendedButton(int xPos, int yPos, int width, int height, Text displayString, PressAction handler) {
+		super(xPos, yPos, width, height, displayString);
+		pressAction = handler;
+		this.label = null;
+		this.shouldRenderTooltip = true;
+	}
+
+	public ExtendedButton(int xPos, int yPos, int width, int height, Text displayString, PressAction handler, @Nullable String label, boolean shouldRenderTooltip) {
+		super(xPos, yPos, width, height, displayString);
+		pressAction = handler;
 		this.label = label;
+		this.shouldRenderTooltip = shouldRenderTooltip;
+	}
+
+	@Override
+	public boolean mouseClicked(double mouseX, double mouseY, int button) {
+		if (this.active && this.visible) {
+			if (button == 1) {
```

Comment:
Did you test this on a Left-Handed mouse configuration?

In this case a "Left-Click" is actually the right mouse button.

Reply:
Tested, works